apps/allocations/app/components/Panel/NewBudget.js

```diff
@@ -0,0 +1,99 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import { DropDown, TextInput } from '@aragon/ui'
+import styled from 'styled-components'
+
+import { Form, FormField } from '../Form'
+import { isStringEmpty } from '../../utils/helpers'
+import { MIN_AMOUNT } from '../../utils/constants'
+
+// TODO:: This should be votingTokens from account?
+const INITIAL_STATE = {
+  name: '',
+  nameError: true,
+  amount: '',
+  amountError: true,
+  currency: 0,
+}
+
+class NewBudget extends React.Component {
+  static propTypes = {
+    onCreateBudget: PropTypes.func.isRequired
+  }
+
+  state =  INITIAL_STATE
+
+  changeField = e => {
+    const { name, value } = e.target
+    this.setState({ [e.target.name]: e.target.value })
+    this.setState({ [name + 'Error']: isStringEmpty(value) ||
+                    (name === 'amount' && value < MIN_AMOUNT) })
+  }
+
+  createBudget = () => {
+    const { name, amount } = this.state
+
+    this.props.onCreateBudget({
+      name,
+      amount,
+    })
+    this.setState(INITIAL_STATE)
+  }
+
+  render() {
+
+    const { name, nameError, amount, amountError, currency } = this.state
+
+    return (
+      <Form
+        onSubmit={this.createBudget}
+        submitText="Create budget"
+        disabled={nameError || amountError}
+      >
+        <FormField
+          required
+          label="name"
+          input={
+            <TextInput
+              name="name"
+              onChange={this.changeField}
+              wide={true}
+              value={name}
+            />
+          }
+        />
+        <FormField
+          required
+          label="amount"
+          input={
+            <InputGroup>
+              <TextInput
+                name="amount"
+                type="number"
+                min={MIN_AMOUNT}
+                onChange={this.changeField}
+                wide={true}
+                value={amount}
+                css={{ borderRadius: '4px 0px 0px 4px' }}
+              />
+              <DropDown
+                name="currency"
+                css={{ borderRadius: '0px 4px 4px 0px' }}
+                items={[ 'ETH', 'DAI' ]}
```

Comment:
Are you sure that we only accept ETH and DAI as currencies? This is a critical limitation to ensure double checking with the specs, anyway it seems a good start and since adding more tokens can be complex, we can deal with that in a separate PR

Reply:
Oh no, not at all. I thought I would just put some sort of stub here and you would take care of that when you do the Web3 integration by pulling a list of currencies from somewhere. I don't think this should be hardcoded on the front end. Or should it?

Reply:
Ok, I think your proposal is fair, since we need kind of web3 elements for this 👍